src/test/groovy/liquibase/ext/neo4j/lockservice/Neo4jLockServiceTest.groovy

```diff
@@ -0,0 +1,95 @@
+package liquibase.ext.neo4j.lockservice
+
+import liquibase.database.Database
+import liquibase.database.DatabaseFactory
+import org.hamcrest.CoreMatchers
+import org.neo4j.driver.AuthTokens
+import org.neo4j.driver.Driver
+import org.neo4j.driver.GraphDatabase
+import org.neo4j.driver.Result
+import org.testcontainers.containers.Neo4jContainer
+import spock.lang.Shared
+import spock.lang.Specification
+
+import java.time.ZonedDateTime
+import java.util.function.Consumer
+import java.util.logging.LogManager
+
+import static org.hamcrest.CoreMatchers.notNullValue
+import static org.hamcrest.Matchers.equalTo
+import static org.hamcrest.Matchers.lessThan
+import static org.junit.Assert.assertThat
+
+class Neo4jLockServiceTest extends Specification {
+
+    static {
+        LogManager.getLogManager().reset();
+    }
+
+    private static final String PASSWORD = "s3cr3t"
+
+    @Shared
+    Neo4jContainer neo4jContainer = new Neo4jContainer()
+            .withAdminPassword(PASSWORD)
+
+    @Shared
+    Driver driver
+
+    def Database database
+
+    def neo4jLockService = new Neo4jLockService()
+
+    def setupSpec() {
+        neo4jContainer.start()
+        driver = driver()
+    }
+
+    def cleanupSpec() {
+        driver.close()
+        neo4jContainer.stop()
+    }
+
+    def setup() {
+        database = DatabaseFactory.instance.openDatabase(
+                "jdbc:neo4j:" + neo4jContainer.getBoltUrl(),
+                "neo4j",
+                PASSWORD,
+                null,
+                null
+        )
+        neo4jLockService.setDatabase(database)
+    }
+
+    def cleanup() {
+        consumeQueryResult("MATCH (n) DETACH DELETE n", {})
+        database.close()
+    }
+
+    def "acquires lock from the underlying database"() {
+        when:
+        def acquiredLock = neo4jLockService.acquireLock()
+
+        then:
+        assertThat acquiredLock, CoreMatchers.is(true)
+        consumeQueryResult(
```

Comment:
@nvoxland please let me know if there is a better way to express the assertions below in Spock style

Reply:
Part of what I like about spock is that it tries to force you to separate your "when" (setup and running logic) and your "then" (assertions) sections. I think that makes the tests more straightforward and easy to read. You can switch back and forth between them as well.

Apply that to your consumeQueryResults, I think you should separate out the "make a query" from the assertion part. As psuedo-code (not trying to make it work), more like:

```
    def "acquires lock from the underlying database"() {
        when:
        def acquiredLock = neo4jLockService.acquireLock()

        then:
        assertThat acquiredLock, CoreMatchers.is(true)

        when:
        def lockNodeProperties = query("MATCH (l:__LiquibaseLock:__LiquigraphLock) RETURN l.id AS id, l.locked_by AS locked_by, l.grant_date AS grant_date")

        then:
        assertThat lockNodeProperties["id"], notNullValue()
        assertThat lockNodeProperties["grant_date"], lessThan(ZonedDateTime.now())
        assertThat lockNodeProperties["locked_by"], equalTo("Neo4jLockService")
    }

```

The "then" potion automatically does the assertion checking nicely by assuming every line is statement that returns true.

So you can simplify it down to:

```
    def "acquires lock from the underlying database"() {
        when:
        def acquiredLock = neo4jLockService.acquireLock()

        then:
        acquiredLock == true

        when:
        def lockNodeProperties = query("MATCH (l:__LiquibaseLock:__LiquigraphLock) RETURN l.id AS id, l.locked_by AS locked_by, l.grant_date AS grant_date")

        then:
        lockNodeProperties["id"] != null
        lockNodeProperties["grant_date"] < ZonedDateTime.now()
        lockNodeProperties["locked_by"] =="Neo4jLockService"
    }
```